pkg/deploy/generator/resources.go

```diff
@@ -556,6 +558,66 @@ func (g *generator) lb() *arm.Resource {
 	}
 }
 
+func (g *generator) lbAlerts() []*arm.Resource {
+	return []*arm.Resource{
+		{
+			Resource: mgmtmonitor.ActionGroupResource{
+				ActionGroup: &mgmtmonitor.ActionGroup{
+					Enabled:        to.BoolPtr(true),
+					GroupShortName: to.StringPtr("lbhealth"),
+				},
+				Name:     to.StringPtr("rp-lb-health-ag"),
+				Type:     to.StringPtr("Microsoft.Insights/ActionGroups"),
+				Location: to.StringPtr("Global"),
+			},
+			APIVersion: apiVersions["monitor"],
+			DependsOn: []string{
+				"[resourceId('Microsoft.Network/loadBalancers', 'rp-lb')]",
+			},
+		},
+		{
+			Resource: mgmtmonitor.MetricAlertResource{
+				MetricAlertProperties: &mgmtmonitor.MetricAlertProperties{
+					Actions: &[]mgmtmonitor.MetricAlertAction{
+						{
+							ActionGroupID: to.StringPtr("[resourceId('Microsoft.Insights/ActionGroups', 'rp-lb-health-ag')]"),
+						},
+					},
+					Enabled:             to.BoolPtr(true),
+					EvaluationFrequency: to.StringPtr("PT5M"), //every 5min
+					Severity:            to.Int32Ptr(4),
+					Scopes: &[]string{
+						"[resourceId('Microsoft.Network/loadBalancers', 'rp-lb')]",
+					},
+					WindowSize:         to.StringPtr("PT15M"), //15min
+					TargetResourceType: to.StringPtr("Microsoft.Network/loadBalancers"),
+					AutoMitigate:       to.BoolPtr(true),
+					Criteria: mgmtmonitor.MetricAlertSingleResourceMultipleMetricCriteria{
+						AllOf: &[]mgmtmonitor.MetricCriteria{{
+							CriterionType:   mgmtmonitor.CriterionTypeStaticThresholdCriterion,
+							MetricName:      to.StringPtr("DipAvailability"),
+							MetricNamespace: to.StringPtr("microsoft.network/loadbalancers"),
+							Name:            to.StringPtr("HealthProbeBelow99"),
+							Operator:        mgmtmonitor.LessThan,
+							Threshold:       to.Float64Ptr(99),
+							TimeAggregation: mgmtmonitor.Average,
+						},
+						},
+						OdataType: mgmtmonitor.OdataTypeMicrosoftAzureMonitorSingleResourceMultipleMetricCriteria,
+					},
+				},
+				Name:     to.StringPtr("healthprobe-alert"),
+				Type:     to.StringPtr("Microsoft.Insights/metricAlerts"),
+				Location: to.StringPtr("global"),
+			},
+			APIVersion: apiVersions["monitor"],
+			DependsOn: []string{
+				"[resourceId('Microsoft.Insights/ActionGroups', 'rp-lb-health-ag')]",
```

Comment:
is it this that depends on rp-lb rather than the action group? if so, put the dependsOn here.